File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/Inheritance/ElectronicProduct.java
class ElectronicProduct extends Product {
    // Additional field
    private String brand;
    private String model;

    public ElectronicProduct(int productId, String name, double price, String manufacturer, String brand, String model) {
        super(productId, name, price, manufacturer); // Call the superclass constructor
        this.brand = brand;
        this.model = model;
    }

    // Getter and Setter for the additional field
    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

        public String getModel() {
            return model;
        }

        public void setModel(String model) {
            this.model = model;
        }

    // Override the displayInfo method to include additional information
    @Override
        public void displayInfo() {
            super.displayInfo(); // Call the superclass method
            System.out.println("Brand: " + brand + ", Model: " + model);
        }
}

File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/inheritance/electronic_product.py
from product import Product

class ElectronicProduct(Product):
    def __init__(self, productId, name, price, manufacturer, brand, model):
        super().__init__(productId, name, price, manufacturer)
        self._brand = brand
        self._model = model

    # Getter and Setter for the additional field
    def getBrand(self):
        return self._brand

    def setBrand(self, brand):
        self._brand = brand

    def getModel(self):
        return self._model

    def setModel(self, model):
        self._model = model

    # Override the displayInfo method to include additional information
    def displayInfo(self):
        super().displayInfo()
        print(f"Brand: {self._brand}, Model: {self._model}")